src/core/server/http/cookie_session_storage.ts

```diff
@@ -139,9 +154,26 @@ export async function createCookieSessionStorageFactory<T>(
     path: basePath,
     clearInvalid: false,
     isHttpOnly: true,
-    isSameSite: false,
+    isSameSite: cookieOptions.sameSite === 'None' ? false : cookieOptions.sameSite ?? false,
   });
 
+  // A hack to support SameSite: 'None'.
+  // Remove it after update Hapi to v19 that supports SameSite: 'None' out of the box.
+  if (cookieOptions.sameSite === 'None') {
+    log.debug('Patching Statehood.prepareValue');
+    const originalPrepareValue = Statehood.prepareValue;
+    Statehood.prepareValue = function kibanaStatehoodPrepareValueWrapper(
+      name: string,
+      value: unknown,
+      options: any
+    ) {
+      if (name === cookieOptions.name) {
+        options.isSameSite = cookieOptions.sameSite;
+      }
+      return originalPrepareValue(name, value, options);
+    };
+  }
```

Comment:
`Statehood.prepareValue` is monkey patched every time the `createCookieSessionStorageFactory` is called, meaning that it could be possible to have multiple wrappers nested if calling the function multiple times. Also due to the `if (cookieOptions.sameSite === 'None')` patching condition, if two calls were to be made to `createCookieSessionStorageFactory` with distinct `cookieOptions.sameSite` values, they would both be affected by the patch.

Note that `httpSetup.createCookieSessionStorageFactory` is only called once in `x-pack/plugins/security/server/authentication/index.ts`, so it effectively probably have no functional impact.

Reply:
> be made to createCookieSessionStorageFactory with distinct cookieOptions.sameSite values, they would both be affected by the patch.

Will they?
There is a guard by cookie name.
```js
if (name === cookieOptions.name) {
  options.isSameSite = cookieOptions.sameSite;
}
```

Reply:
> be made to createCookieSessionStorageFactory with distinct cookieOptions.sameSite values, they would both be affected by the patch.

Will they?
There is a guard by cookie name.
```js
if (name === cookieOptions.name) {
  options.isSameSite = cookieOptions.sameSite;
}
```

Reply:
Probably not close to a real scenario, and probably doesn't make any sense, but:

```ts
cookieOptionsA = {
  name: 'foo',
  sameSite:'None',
}

cookieOptionsB = {
  name: 'foo',
  sameSite:'Lax',
}

const fac1 = createCookieSessionStorageFactory(any, any, cookieOptionsA, any);
const fac2= createCookieSessionStorageFactory(any, any, cookieOptionsB, any);
```

But it seems like the patch would still work with values others than `None` anyway. So the only (minor) concern is that the patch may be applied multiple times, which is still acceptable as `patch(patch(x)) === patch(x)`

Reply:
> So the only (minor) concern is that the patch may be applied multiple times, which is still acceptable as patch(patch(x)) === patch(x)

yeah, that was my reasoning